File: 4. Hard/findTheOddInteger.cpp
/*
This Program will find the number occuring only one time in any array.

Test cases must include a number which occur one time and all the other
numbers will occur for even number of time.

LINK : https://edabit.com/challenge/wgnmQTbfssuhvZHqe
*/


#include<bits/stdc++.h>
using namespace std;

int findodd(vector<int> arr)
{
    int i;
    map<int,int> mp;
    for(i=0;i<arr.size();i++) mp[arr[i]]++;
    for(auto itr=mp.begin();itr!=mp.end();itr++)
    {
        if((itr->second)%2!=0) 
            return itr->first;
    }
}

int main()
{
    int i,j;
    vector<int> arr={1,1,2,2,3,3,4,4,5};
    int odd=findodd(arr);
    cout<<odd;
    return 0;
}

File: 3. Medium/findtheOdd Integer.py

  #This program finds the integer which appears odd number of times
  # Problem : https://edabit.com/challenge/wgnmQTbfssuhvZHqe


def getOddOccurrence(arr, arr_size): 
	
	for i in range(0,arr_size): 
		count = 0
		for j in range(0, arr_size): 
			if arr[i] == arr[j]: 
				count+=1
			
		if (count % 2 != 0): 
			return arr[i] 
		
	return -1
	
	

arr = [2, 3, 5, 4, 5, 2, 4, 3, 5, 2, 4, 4, 2 ] 
n = len(arr) 
print(getOddOccurrence(arr, n)) 

 
